### DISCUSSION TITLE
TreeView - Detect the node on which anything was dropped (drag and drop)

### DISCUSSION THREAD

**Discussion by gautambjain** [microsoft/microsoft-ui-xaml#7882] [Q&A] [answered]
I  using the Drop event to detect when anything is dropped from another ListView.

In a TreeView, how do I know the exact TreeViewNode item on which the drop event occurred?

Please help.

**Comment by castorix** [comment, marked as answer]
You can use _VisualTreeHelper.FindElementsInHostCoordinates_ in Drop event (then DataContext to get Node)

A test with a TextBlock dropped in a TreeView (I used GetCursorPos + ScreenToClient to get coordinates) :

![TreeView_Drop](https://user-images.githubusercontent.com/22345506/199972335-2fe58407-b758-4e8b-a291-473788dfbbdb.gif)

**Reply by gautambjain** [reply]
This answer plus future discussion [in this conversion gives complete solution.](https://github.com/microsoft/microsoft-ui-xaml/discussions/7883) Thanks to @castorix .
